Content.Server/Chemistry/Metabolism/HealthChangeMetabolism.cs

```diff
@@ -0,0 +1,75 @@
+using Content.Server.GameObjects.Components.Nutrition;
+using Content.Shared.Chemistry;
+using Content.Shared.Interfaces.Chemistry;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Serialization.Manager.Attributes;
+using Content.Shared.Damage;
+using Content.Shared.GameObjects.Components.Damage;
+
+
+namespace Content.Server.Chemistry.Metabolism
+{
+    /// <summary>
+    /// Default metabolism for medicine reagents. Attempts to find a DamageableComponent on the target,
+    /// and to update its damage values.
+    /// </summary>
+    [DataDefinition]
+    public class HealthChangeMetabolism : IMetabolizable
+    {
+        /// <summary>
+        /// How much of the reagent should be metabolized each sec.
+        /// </summary> 
+        [DataField("rate")]
+        public ReagentUnit MetabolismRate { get; set; } = ReagentUnit.New(1);
+
+        /// <summary>
+        /// How much damage is changed when 1u of the reagent is metabolized.
+        /// </summary>
+        [DataField("healthChange")]
+        public float HealthChange { get; set; } = 1.0f;
+
+        /// <summary>
+        /// Class of damage changed, Brute, Burn, Toxin, Airloss.
+        /// </summary> 
+        [DataField("damageClass")]
+        public DamageClass DamageType { get; set; } =  DamageClass.Brute;
+        
+
+        /// <summary>
+        /// Remove reagent at set rate, changes damage if a DamageableComponent can be found.
+        /// </summary>
+        /// <param name="solutionEntity"></param>
+        /// <param name="reagentId"></param>
+        /// <param name="tickTime"></param>
+        /// <returns></returns>
+        ReagentUnit IMetabolizable.Metabolize(IEntity solutionEntity, string reagentId, float tickTime)
+        {
+            var metabolismAmount = ReagentUnit.New(MetabolismRate.Float());
+            float accumulatedHealth = 0;
+
+            if (solutionEntity.TryGetComponent(out IDamageableComponent? health))
+            {
+                health.ChangeDamage(DamageType, (int)HealthChange, true);
+                float decHealthChange = (float) (HealthChange - (int) HealthChange);
+                accumulatedHealth += decHealthChange;
```

Comment:
this accumulatedHealth stuff is kind of useless because it's a local variable and doesn't carry state from each Metabolize call to the next; instead, it should be a `private field _accumulatedHealth` on HealthChangeMetabolism (which will allow you to actually accumulate damage)

Reply:
you should doc the accumulatedHealth stuff as well, its a little non-obvious why you're doing it

Reply:
I tested it with decimals, and it worked. But maybe that's because I was the only user of the class.

Reply:
like you tested with healthchange at something like -2.5 and after 2 seconds you had taken 5 damage?

Reply:
I tested .5, 1 and 1.5 and they all worked, but that's because I have some local changes I forgot to push, and can't for a while.

Reply:
local changes are now pushed and should be working just fine.